### ISSUE TITLE
[markers] add problems-widget filter toolbar item

### ISSUE THREAD

**Issue by vince-fugnitto** [theia-ide/theia#5135]
**Description**

Currently, there is no way to filter `problem markers` from the `problems-widget`.
I propose we implement filtering of markers by **error type** or by **text matching** similarly to vscode.

<div align='center'>

<img width="295" alt="Screen Shot 2019-05-13 at 3 37 00 PM" src="https://user-images.githubusercontent.com/40359487/57649051-34ec7780-7595-11e9-90c4-266a862c9443.png">

</div>

**Comment by 15068821183** [comment]
Is there any progress？

**Comment by vince-fugnitto** [comment]
> Is there any progress？

@15068821183 no, no one has worked on the issue as far as I know. If interested, you can try to contribute a pull-request for the feature.
